pkg/didcomm/common/service/service.go

```diff
@@ -35,12 +35,39 @@ type Header struct {
 	Type   string           `json:"@type"`
 }
 
+func (h *Header) clone() *Header {
+	if h == nil {
```

Comment:
This can never happen. The caller would get a panic if they attempted to call this method on a nil reference.

Reply:
It will not panic I have a test for that
https://github.com/hyperledger/aries-framework-go/pull/591/files#diff-2b388021d16ee766a8d669d2e3d88ffeR98-R101

Reply:
We should not be nil checking the receiver pointer. This is overkill and unusual.

Reply:
`Clone` is a public method and we do not know how clients will use it. I just provided a reliable solution instead of panic. We should not expect that the message cannot be nil. If the message is nil and it is possible to call the `Clone` method (according to GO) why we should panic instead of handling it and return nil. Copy <nil> DIDCommMsg returns <nil> DIDCommMsg it does not break the logic right? and more reliable. This `Clone` function has nothing in common with Client usage. If the Client should not work with nill msg it is up to him what to do in such case but it should not be related to us (single responsibility principle) 

clients: packages `didexchange, introduce` 
us: `package service`

Reply:
I'm not concerned about clients getting a panic on calling a method on a nil variable. That is quite normal and expected in Go development :).

Is it impossible to have a nil in our current usage?

Reply:
No, it is not, I can remove it and all tests will pass

Reply:
Do you want me to remove it :)?